### DISCUSSION TITLE
Trying to delete some scenario

### DISCUSSION THREAD

**Discussion by ManuSecsearch** [tetrane/reven-community#24] [Q&A]
Hello guys, 
i'm using the free version 
It happened a replay stopped in some weird stage during the we. I don't know if there was some electric power shutdown or whatever but i'm facing now a situation i'm unable to delete the scenario. i got a red box popup with the following error 'ReplayScenarioTask' Object has no attribute 'replayscenariotask_ptr' and i cannot delete it.
The issues are coming now once i'm trying to setup a new scenario nothing related i got some error even on the replay for memory analyzing.

**Comment by dureuill** [comment]
Hello,

Thank you for your report! From the report, it looks like you are facing two separate issues:

1. Cannot delete a scenario.
2. Cannot replay another scenario.

(1) is a known issue and we're already looking for a fix to introduce in a future version. Meanwhile we can provide you with a way to delete your scenario by dropping to a postgresql shell.

Could you provide more information about the error you encounter in (2), such as the log of the failing memory history replay task (as memory history resources can get rather large, sometimes there is a disk space issue during the replay)?

To fix (1), you will need the ID of the offending scenario that won't get deleted:

1. In the Project Manager, click the "Scenario Manager" tab
2. Click the name of the offending scenario
3. The ID of the scenario is visible in the URL of that scenario (it is an integer, **not** the UUID of the scenario visible on this page)


Once you got your scenario ID, you will need to get to a Postgresql shell on your REVEN installation.

1. Get to the working directory for your REVEN installation.
    a. If using the Docker installation:
        1.  `docker exec -it -u reven $(id -un)-reven bash`
        2.  `cd /reven`
    b. If running the native installation:
        1. `cd path/to/reven-2.10.2-free-mecha-buster`
2. `export QUASAR_SETTING_FILE=$(pwd)/settings.py`
3. `cd share/reven/quasar/`
4. `source venv/bin/activate`
5. `./manage.py dbshell`

From the Postgresql shell, execute the following sql snippet after replacing the `scenario_id` variable with the offending scenario's ID:

```sql
\set scenario_id = YOUR_SCENARIO_ID_HERE

DELETE FROM scenario_replayresourcescenariotask WHERE replayscenariotask_ptr_id IN (SELECT task_ptr_id FROM scenario_replayscenariotask WHERE scenario_id = :scenario_id);
DELETE FROM scenario_replayactionscenariotask WHERE replayscenariotask_ptr_id IN (SELECT task_ptr_id FROM scenario_replayscenariotask WHERE scenario_id = :scenario_id);

DELETE FROM scenario_replayscenariotask_dependencies WHERE from_replayscenariotask_id IN (SELECT task_ptr_id FROM scenario_replayscenariotask WHERE scenario_id = :scenario_id) OR to_replayscenariotask_id IN (SELECT task_ptr_id FROM scenario_replayscenariotask WHERE scenario_id = :scenario_id);

WITH deleted_tasks AS (
    DELETE FROM scenario_replayscenariotask 
    WHERE task_ptr_id IN (SELECT task_ptr_id FROM scenario_replayscenariotask WHERE scenario_id = :scenario_id)
    RETURNING *
)
DELETE FROM task_task WHERE id in (SELECT task_ptr_id FROM deleted_tasks);
```

If the deletion is successful, the snippet should return the number of deleted rows (e.g. `DELETE 42`) for each operation. Please report if it returns any error instead.

Once the offending replay tasks have been deleted from the database, you should be able to delete the scenario from the scenario list.

I hope this message will help you.

**Comment by johdcmlaker2000** [comment]
Have the same issue. The problem started to occur with version 2.10.2, with version 2.10.1 I have never encountered it.

**Reply by dureuill** [reply]
Hey, sorry you're encountering this issue!  Are your issues exactly as described in the original post:
- (A) You are unable to delete some scenario
- (B) You are unable to replay another one?

Regarding (B), can you maybe provide more information about the failing replays on the scenario you can't replay?

Regarding (A):
1. Did you encounter the issue after deleting replay tasks (going to "Tasks & Sessions", then "see all" under "Active tasks", and finally deleting tasks here?
1. Did the manual fix in my answer above help you to get in a state where you can delete the offending scenario again?

If the answer to questions (1) and (2) are "yes", note that we will propose a fix in the next release (not scheduled yet).

Thanks for reporting!
